api/v1alpha1/clusterconfig_types.go

```diff
@@ -220,7 +236,7 @@ type EKSClusterConfigSpec struct {
 
 	// KubeProxy defines the configuration for kube-proxy.
 	// +kubebuilder:validation:Optional
-	KubeProxy *KubeProxy `json:"kubeProxy,omitempty"`
+	KubeProxy *EKSKubeProxy `json:"kubeProxy,omitempty"`
```

Comment:
Will this work? https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/82a4976bf68bd8aed1d00c0fa5bc5b174a5a5c10/api/variables/json.go#L16
with the wrong type https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/82a4976bf68bd8aed1d00c0fa5bc5b174a5a5c10/api/variables/aggregate_types.go#L28

Reply:
Right, and I'm pretty sure `EKSKubeProxy` is only ever going to be a strict subset of `KubeProxy`.

Go clients would understand the Go API which defines the custom type for `EKSKubeProxy` so I don't think this is a problem.